snuba/datasets/plans/storage_plan_builder.py

```diff
@@ -176,6 +179,19 @@ def build_and_rank_plans(
         mappers = storage_connection.translation_mappers
         cluster = self.get_cluster(storage, query, settings)
 
+        # Return failure if storage readiness state is not supported in current environment
+        if snuba_settings.READINESS_STATE_FAIL_QUERIES:
+            assert isinstance(storage, ReadableTableStorage)
+            readiness_state = storage.get_readiness_state()
+            if readiness_state.value not in snuba_settings.SUPPORTED_STATES:
+                cause = StorageNotAvailable(
+                    f"The selected storage={storage.get_storage_key().value} is not available in this environment yet. To enable it, consider bumping the storage's readiness_state."
+                )
+                raise QueryPlanException.from_args(
```

Comment:
Why do you wrap `StorageNotAvailable` in `QueryPlanException` instead of having `StorageNotAvailable` be a subclass of `QueryPlanException`?

Reply:
Just following the pattern we use for QueryException. For example, https://github.com/getsentry/snuba/blob/6582b774e60065065271ef74b4720eef8767f155/snuba/web/query.py#L342-L349

Reply:
I was following the pattern we use for QueryException, but making StorageNotAvailable inherit from `QueryPlanException` simplifies a few type checks. Updated!